students/preactive/Lesson3/list_lab.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python3
+
+
+def series_one():
+    # Create a list that contains “Apples”, “Pears”, “Oranges” and “Peaches”.
+    # Display the list (plain old print() is fine…).
+    fruits = ["Apples", "Pears", "Oranges", "Peaches"]
+    print(fruits)
+
+    # Ask the user for another fruit and add it to the end of the list.
+    # Display the list.
+    response_one = input("Add a fruit:\n>")
+    fruits.append(response_one)
+    print(fruits)
+
+    # Ask the user for a number and display the number back to the user and
+    # the fruit corresponding to that number (on a 1-is-first basis).
+    # Remember that Python uses zero-based indexing,
+    # so you will need to correct.
+    num_fruits = len(fruits)
+    response_two_txt = "Pick a integer from 1 to %s:\n>" % str(num_fruits)
+    response_two = input(response_two_txt)
+    try:
+        response_two_int = int(response_two)
+        if response_two_int not in range(1, num_fruits + 1):
+            print("Not in range. Try again, mouth breather.")
+            series_one()
+        else:
+            print(response_two + ": " + str(fruits[response_two_int - 1]))
+    except:
```

Comment:
It's a best practice to narrow in on a specific exception, so you don't end up allowing some unanticipated error to pass silently. Seems like you'd want to catch ValueError here and modify your error message, since you're already checking the input range inside your try block.

Reply:
will ValueError catch when someone tries to inject funny business like: i̷̡̫̠͖͌̾̑́̑̀ͨͦ͛̚͠͞n̶͙̰̭̳̖̘̹̻͚̜̾ͣ̔ͮͤ̉͂̐̌̋ͪ̀̌́͜p̸̛͍̯̯͖̲̹̜̦̼̲̥͌͌̇͗̆̾ͬ̏̀ͪ͆̅ͧ̍͗̍̉̄̓͘̕͢ͅu̽̇̋̋̊̃ͪ̂̓̾ͣ̆͂̈́́͏̢̰̫̭̟̣̞̗̰̰̼͇̤͉̯̹̣̠̗͠t̸̝͔̥̲͈̼͈̦̰̭̼̗͔̹̳̭̦͎͒͆ͦ̕

Reply:
I think you might want to use a custom exception class for that ;)

Reply:
ok went with ValueError.  Fixed